### PR TITLE
feat: use FormBuilder for access form

### PR DESCRIPTION
**Pull request by thesanjeevsharma**
Related to https://github.com/ghostfolio/ghostfolio/issues/2350

I've made the changes to the Access Form as well.

### CODE REVIEW

File: apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.html [hunk 1]
```diff
@@ -1,35 +1,39 @@
...
+  [formGroup]="accessForm"
+  (keyup.enter)="accessForm.valid && onSubmit()"
+  (ngSubmit)="onSubmit()"
+>
+    <h1 i18n mat-dialog-title>Grant access</h1>
+    <div class="flex-grow-1 py-3" mat-dialog-content>
+      <div>
+        <mat-form-field appearance="outline" class="w-100">
+          <mat-label i18n>Alias</mat-label>
+          <input
```
Comment (dtslvr) [inline apps/client/src/app/components/user-account-access/create-or-update-access-dialog/create-or-update-access-dialog.html:12]: Add `(keydown.enter)="$event.stopPropagation()"`

File: apps/api/src/app/access/create-access.dto.ts [hunk 2]
```diff
@@ -8,4 +8,8 @@ export class CreateAccessDto {
   @IsOptional()
   @IsString()
   granteeUserId?: string;
+
+  @IsOptional()
+  @IsString()
+  type?: string;
```
Comment (dtslvr) [inline apps/api/src/app/access/create-access.dto.ts:14]: `type?: 'PUBLIC';`

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thanks for the PR @thesanjeevsharma. I have added some comments.

**Comment by thesanjeevsharma** [comment]
@dtslvr I've addressed the comments.

**Pull request merged**